test/l1-resources/stream-connection/index.test.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { App, Stack } from "aws-cdk-lib";
+import { Template } from "aws-cdk-lib/assertions";
+import {
+  CfnStreamConnection,
+  CfnStreamConnectionPropsType,
+} from "../../../src";
+
+const RESOURCE_NAME = "MongoDB::Atlas::StreamConnection";
+const PROFILE = "default";
+const PROJECT_ID = "testProjectId";
+const INSTANCE_NAME = "testInstanceName";
+const CONNECTION_NAME = "testConnectionName";
+
+test("AtlasStreamConnection construct should contain default properties", () => {
+  const mockApp = new App();
+  const stack = new Stack(mockApp);
+
+  new CfnStreamConnection(stack, "stream-connection-testing-stack", {
+    profile: PROFILE,
+    instanceName: INSTANCE_NAME,
+    projectId: PROJECT_ID,
+    connectionName: CONNECTION_NAME,
+    type: CfnStreamConnectionPropsType.CLUSTER,
+  });
+
+  const template = Template.fromStack(stack);
+
+  template.hasResourceProperties(RESOURCE_NAME, {
+    Profile: PROFILE,
+    InstanceName: INSTANCE_NAME,
+    ProjectId: PROJECT_ID,
+    ConnectionName: CONNECTION_NAME,
+    Type: CfnStreamConnectionPropsType.CLUSTER,
+  });
```

Comment:
for this unit test can we include all properties? Just to make sure they are all being mapped correctly to the CFN template.

Reply:
yes I think we can have 2 unit tests here for each Kafka & Cluster connection type

Reply:
Done!